src/components/ContactUsForm/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useState } from 'react';
+import Input, { InputType } from '@components/Input';
+import TextArea from '@components/TextArea';
+import InputSelect from '@components/InputSelect';
+import Button from '@components/Button';
+
+const SUBJECT_OPTIONS = [
+  {
+    label: '-- Select a subject --',
+    value: ''
+  },
+  {
+    label: 'Press Inquiry',
+    value: 'press-inquiry'
+  },
+  {
+    label: 'Signup Issue',
+    value: 'account-issues'
+  },
+  {
+    label: 'Other',
+    value: 'other'
+  }
+];
+
+const INITIAL_STATE = {
+  name: '',
+  email: '',
+  subject: '',
+  message: ''
+};
+
+const ContactUsForm = () => {
+  const [data, setData] = useState(INITIAL_STATE);
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleChange = (key: string, value: string) => {
+    setData((d) => ({ ...d, [key]: value }));
```

Comment:
`setState()` automatically merges a partial state into the current state, we only needed to call it with the changed parts. https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged

Reply:
@duranmla not for functional components https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

![image](https://user-images.githubusercontent.com/20747535/115281382-8cdf7180-a10e-11eb-9c3f-8fb52a0cb448.png)

Reply:
Sorry for the confusion then! Roger that!